src/main/java/org/checkerframework/specimin/UnsolvedSymbolVisitor.java

```diff
@@ -176,6 +178,16 @@ public class UnsolvedSymbolVisitor extends ModifierVisitor<Void> {
   /** New files that should be added to the list of target files for the next iteration. */
   private final Set<String> addedTargetFiles = new HashSet<>();
 
+  /** Stores the set of method names declared in the currently visiting class. */
+  private final Set<String> declaredMethod = new HashSet<>();
```

Comment:
Rather than names, should this set contain signatures? I'm thinking of a test case where there is an unsolved superclass/superinterface that declares two overloads of the same method that have different parameter lists, and the target method uses both. For example, something like:
```
void target(boolean b) {
  if (b) { this.foo(1); } else { this.foo(1, "random string"); }
}
```

Please 1) write such a test case, and then 2) consider whether how set should handle overloads.

Reply:
Professor,

Regarding the `declaredMethod` field, it only records methods declared in the current class. This allows us to recognize methods from superclasses or superinterfaces, even if they are implicitly called, such as `this.foo()` or `baz()` (by checking if `foo` or `baz` is already declared in the current class). Hence, for your test case, `declaredMethod` remains effective.

But your observation raises an important point about the handling method overloads . If `foo(int)` is declared in the current class, and there is an implicit call for a method from a superclass, such as `foo("hello")`, Specimin will not be able to realize that` foo("hello")` is not declared in the current class because it only relies on method name. So I will update Specimin and add corresponding test case.